src/exit_status.h

```diff
@@ -15,10 +15,30 @@
 #ifndef NINJA_EXIT_STATUS_H_
 #define NINJA_EXIT_STATUS_H_
 
-enum ExitStatus {
-  ExitSuccess,
+// The underlying type of the ExitStatus enum, used to represent a platform-specific
+// process exit code.
+#ifdef _WIN32
+#define EXIT_STATUS_TYPE unsigned long
+#else  // !_WIN32
+#define EXIT_STATUS_TYPE int
+#endif  // !_WIN32
+
+// The platform-specific value of ExitInterrupted returned by Ninja in case of use interruption (e.g. Ctrl-C).
+// On Windows, the convention used by the C runtime library is to return 3, even
+// though the system error `ERROR_CONTROL_C_EXIT` is 572. Some applications also return 1
+// which is ExitFailure. The value 2 is chosen to preserve the existing Ninja behavior on this
+// platform. On Posix, this is simply 128 + SIGINT. Note that Ninja will map SIGHUP
+// and SIGTERM interrupted to ExitInterrupted as well.
+#ifdef _WIN32
+#  define EXIT_INTERRUPTED_VALUE  2
+#else  // !_WIN32
+#  define EXIT_INTERRUPTED_VALUE  130
+#endif  // !_WIN32
+
+enum ExitStatus : EXIT_STATUS_TYPE {
+  ExitSuccess=0,
   ExitFailure,
-  ExitInterrupted
+  ExitInterrupted=EXIT_INTERRUPTED_VALUE,
```

Comment:
Since the exit code used to be [130](https://github.com/ninja-build/ninja/blob/b78335e0f960bb547c208254e65c9a20fcd4da29/src/ninja.cc#L1578-L1583) for all the OS, should we make it 130 here as well, maybe?

Reply:
Pushed `ExitInterrupted=130`. Please let me know if you have a better idea regarding it.

Reply:
Oh, my mistake for suggesting otherwise. Yes, I would be in favor of keeping returning 130 on Windows if that's what we already do (I guess nobody cares though).